sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -35,12 +34,18 @@ namespace Azure.Messaging.EventHubs.Producer
     ///
     public class EventHubProducerClient : IAsyncDisposable
     {
+        /// <summary>The maximum number of attempts that may be made to get a <see cref="TransportProducer"/> from the pool.</summary>
+        internal const int MaximumCreateProducerAttempts = 3;
+
         /// <summary>The minimum allowable size, in bytes, for a batch to be sent.</summary>
         internal const int MinimumBatchSizeLimit = 24;
 
         /// <summary>The set of default publishing options to use when no specific options are requested.</summary>
         private static readonly SendEventOptions DefaultSendOptions = new SendEventOptions();
 
+        /// <summary>Sets how long a dedicated <see cref="TransportProducer"/> would sit in memory before its <see cref="TransportProducerPool"/> would remove and close it.</summary>
+        private static readonly TimeSpan PartitionProducerLifespan = TimeSpan.FromMinutes(5);
```

Comment:
@jsquire we have 2 constants that dictate how long a ```PoolItem``` should sit in memory:
- ```EventHubProducerClient.PartitionProducerLifespan```, 5 mins
- ```PoolItem.DefaultRemoveAfterDuration```, 10 mins

I wonder if we should remove one or make the values the same?

Reply:
IIRC, we did this intentionally.    The client will attempt to extend a transport producer if it falls under the 5 minutes.   While, the pool will remove a transport producer if it hasn't been touched in 10 minutes.   The difference in timing reduces the potential for a race condition where the client attempts to extend at the same time the pool is expunging it as unused.

Reply:
Hang on, I know what's happening here.

We created it after @kinelski kindly suggested:

> Don't you think that moving the DefaultRemoveAfterDuration to the TransportProducerPool class would make more sense? It's not being used by the PoolItem class at all. It feels like it's something completely controlled by the caller. We would also want to make it configurable.

I put it in the `SendEventOptions` and you flagged me:

>This isn't something that we want to expose as "when sending" option; this is an implementation detail to the producer client. If we see a future need, we'd want to expose this as part of the producer client options, but for now, I'd prefer to go with a private value.

I then put it as a constant and took the value of 5 minutes from your sketch. Without knowing that the purpose you were foreseeing for that constant was different.

Reply:
I'm questioning myself now.  We should probably flag taking a look at this a bit more during testing.   I *think* we did it intentionally, but it may have just been too many moving parts over a long period of time.   When we move into some stress testing, it should help us to play with things a bit and see how real my concern was.

Reply:
> When we move into some stress testing

Fantastic, look forward to it. Apologies for the naive question, is that something we are going to do in the integration tests?

Reply:
Not a naïve question at all... we're in an odd place with stress testing at the moment.  We've got an informal client for sustained runs that we've been using and have some shims that allow us to plug into the stress environment that the service team uses.   Longer term, the SDK engineering team is looking to build out a stress platform for each of the service libraries to plug into.

For this, I'm thinking that once we get to a point where we think things are ready, either I'll point you at our informal client and ask that you do a couple of 24-72 hour runs, or I'll grab your changes and do one.   Maybe both.

Reply:
That sounds interesting.